### DISCUSSION TITLE
Consider using kloudless to support all cloud services at once

### DISCUSSION THREAD

**Discussion by adriancuadrado** [sboesen/remotely-sync#72] [Ideas]
I have understood they have a library precisely for that that you might want to checkout. I'm on mobile and I don't have the link, but I'll bring it later.

Edit: Here is the link: https://kloudless.com/products/cloud-storage-api/. It's not a library, my bad. It's actually a rest api. Still worth checking out I think.

**Comment by kadisonm** [comment]
That's interesting, I wonder how it would handle something like #67 between the different cloud services.

**Comment by sboesen** [comment]
Definitely interesting, but I think (right now) I would rather slowly integrate remote clouds than add a hard dependency to a third party API. Is there a provider you're missing right now other than [Google Drive and iCloud](#35)?

**Reply by adriancuadrado** [reply]
I'm not even missing Google Drive actually, because Google Drive for desktop already syncs your files anyways. For Android I use FolderSync, so I'm well served.
It's a shame this functionality doesn't come as a library but as a third party API, I understand if you don't want to add such dependency.

Would you consider extracting the part of the code that deals with each cloud provider as an external reusable npm package for others to use? I've been searching for something like this to make a suggestion in this repository and was quite surprised no one worked on something like this before except kloudless.

**Reply by sboesen** [reply]
I think Google has a library, haven't explored it much but looks like it would work [at a glance](https://github.com/googleapis/google-api-nodejs-client). I think we just need code to figure out how to integrate this with our sync strategy. Let me know if I misunderstood here!

**Reply by kadisonm** [reply]
> Would you consider extracting the part of the code that deals with each cloud provider as an external reusable npm package for others to use?

That's an interesting idea but the plugin might be too integrated with Obsidian and could be out of scope. Unless the plugin itself used the npm package then I could see it being useful? @sboesen

**Reply by sboesen** [reply]
Yeah the helpful parts of each of these are already third party libraries, so I don't think this would be helpful but I may be misunderstanding

**Reply by kadisonm** [reply]
Right makes sense, that's what I was thinking about too. I guess an advantage would still be using common methods etc between services but it seems too out of scope for it to make sense for this plugin. Interesting idea though.
